pom.xml

```diff
@@ -569,6 +569,56 @@
                 <jni.classifier>${cuda.version}-arm64</jni.classifier>
             </properties>
         </profile>
+        <profile>
+            <id>bloopInstall</id>
+            <activation>
+                <property>
+                    <name>bloopInstall</name>
+                    <value>true</value>
+                </property>
+            </activation>
+            <properties>
+                <!-- Metals looks at the repo root. so to accomodate scala 2.13  define -->
+                <bloop.configDirectory>${spark.rapids.source.basedir}/.bloop</bloop.configDirectory>
+                <skipTests>true</skipTests>
+                <maven.scaladoc.skip>true</maven.scaladoc.skip>
+                <maven.scalastyle.skip>true</maven.scalastyle.skip>
+            </properties>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-enforcer-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>enforce-bloop-rules</id>
+                                <goals><goal>enforce</goal></goals>
+                                <configuration>
+                                    <rules>
+                                        <requireJavaVersion>
+                                            <message>Metals semanic database requires JAVA_HOME pointinting to JDK 11+, actual Java version: ${java.version}</message>
+                                            <version>[11,)</version>
+                                        </requireJavaVersion>
+                                    </rules>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
+                        <groupId>ch.epfl.scala</groupId>
+                        <artifactId>bloop-maven-plugin</artifactId>
+                        <version>2.0.0</version>
+                        <executions>
+                            <execution>
+                                <id>generate-bloop-projects</id>
+                                <goals><goal>bloopInstall</goal></goals>
+                                <phase>install</phase>
```

Comment:
Is it really necessary to run **all** install goals when completing `bloopInstall`? We also do this in the existing script, but the `ch.epfl.scala:bloop-maven-plugin:bloopInstall` could feasibly be run independently of `mvn install`.

My understanding is that the Bloop server does some building of its own - not clear to me that running an independent `install` step is required.

Reply:
Generally you are right. It is somewhat complicated in spark-rapids. If we want test classes to be included we need at least generate-test-sources phase in spark-rapids. This is past the phases where we do scala-compile-first. Some modules like integration_tests are downstream of the dist module that has a custom package logic. In summary, just to make sure that bloopInstall works robustly for all situations I chose install. 

However, we can make it a property for expert users with the default value `install`